tests/beta/test_beta_qnode.py

```diff
@@ -0,0 +1,900 @@
+            return qml.expval(qml.PauliZ(0))
+
+        qn = QNode(func, dev)
+
+        x = 0.12
+        y = 0.54
+
+        res = qn(x, y)
+
+        assert isinstance(qn.qtape, JacobianTape)
+        assert len(qn.qtape.operations) == 3
+        assert len(qn.qtape.observables) == 1
+        assert qn.qtape.num_params == 2
+
+        expected = qn.qtape.execute(dev)
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+        # when called, a new quantum tape is constructed
+        old_tape = qn.qtape
+        res2 = qn(x, y)
+
+        assert np.allclose(res, res2, atol=tol, rtol=0)
+        assert qn.qtape is not old_tape
+
+    def test_jacobian(self, tol):
+        """Test the jacobian computation"""
+        dev = qml.device("default.qubit", wires=2)
+
+        def func(x, y):
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            qml.CNOT(wires=[0, 1])
+            return qml.probs(wires=0), qml.probs(wires=1)
+
+        qn = QNode(func, dev, diff_method="finite-diff", h=1e-8, approx_order=2)
+        assert qn.gradient_kwargs["h"] == 1e-8
+        assert qn.gradient_kwargs["approx_order"] == 2
+
+        jac = qn.gradient_fn(qn)(0.45, 0.1)
+        assert jac.shape == (2, 2, 2)
+
+    def test_returning_non_measurements(self):
+        """Test that an exception is raised if a non-measurement
+        is returned from the QNode."""
+        dev = qml.device("default.qubit", wires=2)
+
+        def func(x, y):
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            qml.CNOT(wires=[0, 1])
+            return 5
+
+        qn = QNode(func, dev)
+
+        with pytest.raises(
+            qml.QuantumFunctionError, match="must return either a single measurement"
+        ):
+            qn(5, 1)
+
+        def func(x, y):
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(0)), 5
+
+        qn = QNode(func, dev)
+
+        with pytest.raises(
+            qml.QuantumFunctionError, match="must return either a single measurement"
+        ):
+            qn(5, 1)
+
+    def test_inconsistent_measurement_order(self):
+        """Test that an exception is raised if measurements are returned in an
+        order different to how they were queued on the tape"""
+        dev = qml.device("default.qubit", wires=2)
+
+        def func(x, y):
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            qml.CNOT(wires=[0, 1])
+            m = qml.expval(qml.PauliZ(0))
+            return qml.expval(qml.PauliX(1)), m
+
+        qn = QNode(func, dev)
+
+        with pytest.raises(
+            qml.QuantumFunctionError,
+            match="measurements must be returned in the order they are measured",
+        ):
+            qn(5, 1)
+
+    def test_consistent_measurement_order(self):
+        """Test evaluation exceeds as expected if measurements are returned in the

```

Comment:
Do you mean "exceeds" here? Possibly a typo for "proceeds" or "executes"?

Reply:
Oops 😆 
```suggestion
        """Test evaluation proceeds as expected if measurements are returned in the
```